File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 7/29/2026, 10:13:36 AM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        L, R = 0, 1
        maxProfit = 0
        while R < len(prices):
            if (prices[R] - prices[L]) <= 0:
                L = R
                R += 1
            else: 
                maxProfit = max(maxProfit, prices[R] - prices[L])
                R += 1
        return maxProfit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 7/29/2026, 10:13:38 AM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) return 0;

        int l = 0, r = 1, maxTrade = 0;
        while(r < prices.size()){
            if(prices[l] > prices[r]){
                maxTrade = std::max(prices[r] - prices[l], maxTrade);
                l = r;
            }
            else{
                maxTrade = std::max(prices[r] - prices[l], maxTrade);
                r++;
            }

        }
        return maxTrade;
    }
};